okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RemotingOkHttpCall.java

```diff
@@ -274,13 +263,38 @@ public Void visit(QosException.Unavailable exception) {
                 } else {
                     log.debug("Rescheduling call after backoff",
                             SafeArg.of("backoffMillis", backoff.get().toMillis()));
-                    scheduleExecution(callback, backoff.get());
+                    redirectToUrl(
+                            callback,
+                            call,
+                            // Redirect to the "next" URL, whichever that may be, after backing off.
+                            () -> urls.redirectToNext(request().url()),
+                            "Failed to determine valid redirect URL for base URLs " + urls.getBaseUrls(),
+                            request -> scheduleExecution(() -> client.newCallWithMutableState(request,
+                                    backoffStrategy, maxNumRelocations - 1).enqueue(callback), backoff.get()));
```

Comment:
don't think we need to decrease maxNumRelocations in this case. backoff already limits the number of retries for 503s.